src/ASM/ASMs3DLag.C

```diff
@@ -947,11 +947,14 @@ int ASMs3DLag::findElement (double u, double v, double w,
     return -1;
   }
 
-  const std::array<std::pair<double,int>,3> knot {{
-    {u, svol->basis(0).knotInterval(u)},
-    {v, svol->basis(1).knotInterval(v)},
-    {w, svol->basis(2).knotInterval(w)}
-  }};
+  std::array<std::pair<double,int>,3> knot;
+
+#pragma omp critical
+  {
+    knot[0] = {u, svol->basis(0).knotInterval(u)};
+    knot[1] = {v, svol->basis(1).knotInterval(v)};
+    knot[2] = {w, svol->basis(2).knotInterval(w)};
+  }
```

Comment:
Will not not work just to insert the `#pragma` before L950? Since it is just a single statement anyway, or..?